### PR TITLE
Full db implementation

### PR DESCRIPTION
**Pull request by caydenreynolds**

### CODE REVIEW

**Comment by caydenreynolds** [comment]
Mongo stuff occasionally fails, but I can't reproduce the problem consistently enough to debug it easily. We'll see if it becomes a problem

File: src/test/database/MongoConn.test.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                expect(result.foo == "bar" || result.num == 42).toBe(true);
+            });
+        } finally {
+            done();
+        }
+    }
+    test();
+});
+
+test("Sucessfully inserts one object", done => {
```
Comment (ScratchnSniff0) [inline src/test/database/MongoConn.test.js]: Sucessfully -> Successfully

File: src/test/database/MongoConn.test.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            expect(await oneKey).toMatchObject({subobj: expectedSelectResult.subobj, id: expectedSelectResult.id});
+            expect(await byId).toMatchObject(expectedSelectResult);
+        } finally {
+            done();
+        }
+    };
+    test();
+});
+
+test("Sucessfully executes advanced queries", done => {
```
Comment (ScratchnSniff0) [inline src/test/database/MongoConn.test.js]: Sucessfully -> Successfully

File: src/database/IDBConn.js [hunk 3]
```diff
@@ -39,20 +39,39 @@ export default class IDBConn {
...
-     * @param {String} table The table to delete from
+     * Returns an object that indicates a given field is not set in the database
+     * usage example: 
+     *      user.email = conn.not_set();
+     *      let db_user = conn.selectOne(user);
+     * @returns {object} An object that checks for fields that are not set
      */
-    delete(queryObj, table) { throw Exception("Unimplemented"); }
+    not_set() { throw new Error("Unimplemented"); }
 }
```
Comment (ScratchnSniff0) [inline src/database/IDBConn.js]: Is newline at the end of a file idiomatic for JS?

File: src/database/MongoConn.js [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+ * @param {object} obj  The object to mongoize
+ * @returns {object} The mongoized object
+ */
+function _mongoize(obj) {
+    if ('id' in obj) {
+        obj._id = mongodb.ObjectID(obj.id);
+        delete obj.id;
+    }
+    return obj;
+}
```
Comment (ScratchnSniff0) [inline src/database/MongoConn.js]: Is newline at the end of a file idiomatic for JS?

File: src/model/SimObj.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        conn.replace({id: this.id}, this.toJsonObject(), this.tablename);
+    }
+
+    /**
+     * Returns whether or not this SimObj can be modified by user
+     * @param {mode.User} user The user trying to modify this SimObj
+     * @returns {bool} True if user can modify this SimObj, or false otherwise
+     */
+    async modifyableBy(user) { throw new Error("Unimplemented"); }
+}
```
Comment (ScratchnSniff0) [inline src/model/SimObj.js]: Is newline at the end of a file idiomatic for JS?

File: src/test/database/MongoConn.test.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            // delete result._id;
+            expect(result).toMatchObject(updatedObj);
+        } finally {
+            done();
+        }
+    }
+    test();
+});
+
+test("Sucessfully replaces one object", done => {
```
Comment (ScratchnSniff0) [inline src/test/database/MongoConn.test.js]: Sucessfully -> Successfully

File: src/test/database/MongoConn.test.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+            let result = await client.db(DB_NAME).collection(insertTable).findOne(findObj);
+            expect(result).toMatchObject(insertObj);
+        } finally {
+            done();
+        }
+    };
+    test();
+});
+
+test("Sucessfully updates one object", done => {
```
Comment (ScratchnSniff0) [inline src/test/database/MongoConn.test.js]: Sucessfully -> Successfully

File: test.sh [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (ScratchnSniff0) [inline test.sh]: Scary! Share file personally instead?

**Review by ScratchnSniff0 — CHANGES_REQUESTED** [review]
Commented on a few things.  But otherwise looks good to me!

**Review by rdoster13 — APPROVED** [review]
Looks good to me

**Pull request merged**